### PR TITLE
Adding attenuation versus extinction docs

### PR DESCRIPTION
**Pull request by karllark**
Addresses #38. In conjunction with a companion issue karllark/dust_extinction#47.

Docs in dust_attenuation focus on fully explaining attenuation with minimal extinction discussion. Mainly with a pointer to the corresponding discussion in dust_extinction.

### CODE REVIEW

**Comment by karllark** [comment]
@dcorre  : if you are up for reviewing this (and the companion pull request in dust_extinction) that would be great.  I've also asked for a review from two other people - see the dust_extinction issue this pull request addresses.  But I would very much like to have an "experts" eyes on this documentation, especially as the topic of extinction versus attenuation is often not well understood.

**Comment by dcorre** [comment]
Sure, I will be happy to do it. I'll send you my comments by next week.

**Comment by karllark** [comment]
Thanks for the review, I will respond this week (I hope).

File: docs/dust_attenuation/attenuation.rst [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+Attenuation versus Extinction
+#############################
+
+Attenuation
+===========
+
+Dust attenuation refers to the general impact on the spectrum of an object due
+to the presence of dust.  In general, attenuation is used to indicate that the
+geometry of the sources and dust in a system is more complex than a single star
+with a foreground screen of dust.  Examples of such systems include dusty
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: Good point.  I've added a few more words than you suggested to try and make it as clear as possible on the differences.

File: docs/dust_attenuation/attenuation.rst [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+galaxies (composed of many stars) and stars with circumstellar dust.
+
+Measuring attenuation for an object is nominally as straightforward as measuring
+extinction.  The spectrum of the object with dust is compared to an equivalent
+object with no dust.  In reality, this is quite hard as the usual objects for
+such work are galaxies and no two galaxies are similar enough.  This can and has
+been done for samples of galaxies (e.g., starbursts: Calzetti et al. 2000) or by
+using a model for the stellar populations in that galaxy.
+
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: I think this is too much detail for a general description of attenuation.  The goal here is to introduce people to the concept, not give all the possible details.  I don't want to "scare" people off with too much detail.  Maybe there would be a better place for this - possibly in the how to choose a model docs (that still need to be written)?

I do like the additional references and I have added these to the existing test.

I would distinguish between the ways that people measure curves (empirically or with an analytical recipe [e.g., stellar pops + simple attenuation shapes]) and how they can be theoretically calculated from physics (radiative transfer models).  The same way I would distinguish between the ways to measure dust *extinction* curves (e.g., pair method, w/ stellar atmosphere models) and theoretically calculate them from physics (e.g., Mie theory and size distributions of carbonaceous and silicate grains).

File: docs/dust_attenuation/attenuation.rst [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+Attenuation includes the effects of having multiple sources extinguished by
+different columns of dust and the scattering of photons into the observation
+beam.  Unlike extinction, these two effects mean that the attenuation is not
+directly proportional to the amount dust in the system. Hence the ratio of
+attenuations at two different wavelengths *varies* with the amount total system
+dust. This is illustrated below with `WG00` shell, clumpy, mw models where the
+left plot shows the total attenuation as a function of wavelength and the right
+plot shows the same curves normalized by Att(V).
+
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: I like most of the change, but not the last sentence.  This restates the main sentence, but is not as clear as just showing the plot.  Hence, I have left it out and gone directly to the plot.

File: docs/dust_attenuation/attenuation.rst [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+Interstellar dust extinction is the result of photons being absorbed or
+scattered *out* of the line-of-sight by dust grains.  Extinction is
+explicitly linked to the specific geometry of a single star observed
+through a column of dust.
+
+Both dust absorption and scattering out of the line-of-sight are processes
+that are directly proportional to the amount of dust along the line-of-sight.
+As a result, the ratio of dust extinctions at two different wavelengths
+does not vary with different amounts of dust.  This makes the measurement
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: Done.

File: docs/dust_attenuation/attenuation.rst [hunk 5]
```diff
@@ -0,0 +1,150 @@
+.. _AttvsExt:
+
+#############################
+Attenuation versus Extinction
+#############################
+
+Attenuation
+===========
+
+Dust attenuation refers to the general impact on the spectrum of an object due
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: Good point.  I've revised the first part to be more general that what you have suggested as attenuation can extend from the X-ray to mid- and far-IR and beyond.  Attenuation is the impact of dust absorption and scattering *minus* the dust emission.  And re-emission of the absorbed energy happens mainly in the IR, but energy is also emitted in the optical (e.g., ERE) and probably other wavelengths - just not through thermal emission - e.g., photo-luminescence.

File: docs/dust_attenuation/attenuation.rst [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+to the presence of dust.  In general, attenuation is used to indicate that the
+geometry of the sources and dust in a system is more complex than a single star
+with a foreground screen of dust.  Examples of such systems include dusty
+galaxies (composed of many stars) and stars with circumstellar dust.
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: Is this definition needed for this discussion?  I do not think so.

File: docs/dust_attenuation/attenuation.rst [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+total flux from the system due to this scattered flux.  The wavelength where the
+scattered flux peaks is a combination of the total optical depth of the system
+and the impact of multiple scatterings. Note that the total attenuation as
+measured in the V band is always less when the contribution from scattered flux
+into the observer's line-of-sight is included (numbers given in the legends).
+
+.. image:: ./plots/2star_mix_dirty.png
+   :scale: 50 %
+   :align: center
+
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: I've added in most of this suggested text.  But not the one about local galaxies - such work is more specific and not as extensive.  The first sentence is nice as the focus of such work is on the dust radiative transfer effects - and this is what is being discussed in this section.

File: docs/dust_attenuation/attenuation.rst [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+References
+----------
+
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: Done.

File: docs/dust_attenuation/attenuation.rst [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+Calzetti et al. 2000, ApJ, 533, 682
+
+Gordon et al. 2001, ApJ, 551, 269
+
+Gordon et al. 2017, ApJ, 551, 277
+
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: Done.

File: docs/dust_attenuation/attenuation.rst [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+Misselt et al. 2001, A&A, 603, A114
+
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: Done.

File: docs/dust_attenuation/attenuation.rst [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+.. image:: ./plots/2star_mix_dirty.png
+   :scale: 50 %
+   :align: center
+
+References
+----------
+
+Calzetti et al. 2000, ApJ, 533, 682
+
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_attenuation/attenuation.rst]: Done.

**Comment by karllark** [comment]
@dcorre : Hopefully my changes and comments make sense and are ok with you.

**Comment by karllark** [comment]
Merging.  Please raise an issue if there is more work needed.  Just want to get this into circulation and find out what others think.

File: docs/dust_attenuation/attenuation.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (dcorre) [reply docs/dust_attenuation/attenuation.rst]: I used the term attenuation curve throughout my suggestion so I needed to define it. If you do not use this term, it is indeed useless for the discussion

**Comment by dcorre** [comment]
I agree with all of your changes.
You are right about the distinction of the methods to measure the attenuation curve, this is too specific and detailed, and that it should be written in the _how to choose a model_ section (need to find the time).

**Pull request merged**